### DISCUSSION TITLE
Help, desperately want to downgrade/revert Rockstar Launcher version of GTA V

### DISCUSSION THREAD

**Discussion by JokerOmega1** [scripthookvdotnet/scripthookvdotnet#1485] [Q&A]
So these latest two GTA V updates have really screwed up a lot of the script mods I have been using (for example heists mods), they were working just fine before the update, but now almost all of them initiate a script error at some point or another. I have spent hours upon hours searching and trying out different ways to revert my Rockstar Launcher copy of the game, but no method I come across seems to work any more.

Whatever Rockstar did with the last couple of updates really messed things up, at this point I wish they'd stop updating GTA V.

Does anyone know of a way I can revert the Rockstar version of GTA V to a late 2023/early 2024 version of the game? I'd really rather not have to purchase the game again through Steam. I only really kept playing the game thanks to wonderful mods we have available that add so much to the single player experience, now I almost don't want to play anymore given almost all of the script mods I used are broken now.

**Comment by nomakewan** [comment]
They will never stop updating GTAV, so you're gonna have to deal.

If you want to revert, make sure you have version 1.0 (not 2.0!!) of `GTAV_NoLauncher` from 5mods installed to your GTAV folder, and make sure you have automatic updates disabled for GTAV in RGL's settings. Then go ahead and downgrade, and when you want to run the game, start it from the `GTAV_NoLauncher.exe` file instead.

If you're downgrading to a version of GTAV even older than 2824, make sure you also have [kagikn's SocialClub bypass](https://www.gta5-mods.com/tools/exe-integrity-bypass-against-rgl).

**Reply by JokerOmega1** [reply]
That was one of the things I tried a couple of weeks ago, but that method doesn't seem to work anymore as it gives an error message when loading into single player. There are also no options in RGL's settings that disables automatic updates (at least as far as I can tell).

**Reply by nomakewan** [reply]
That's very strange; I have multiple users of BTTFV who followed those same directions and were able to successfully downgrade to v2944 on RGL. I'm not sure why it isn't working for you.

**Reply by JokerOmega1** [reply]
I don't know what to say then, I have searched all through the Rockstar Launcher options, and there is no disable automatic game updates option, if there was I would've done this a long time ago. And with the GTAV_NoLauncher mod, it just gives me a constant "please sign into Rockstar games" message when loading into single player with it, resulting in one having to close the game via alt-f4.

**Reply by JokerOmega1** [reply]
Tried it again, same issues, "Disconnected, please sign into Rockstar Games", you have the option to "reconnect" or "quit", whenever I hit "reconnect" it fails and the only other option is to quit. It's beyond frustrating at this point. I downgraded to game version 1.0.3095.0 with the version of scripthook that corresponds to stated game version, and just doesn't work.

I followed the instructions in this video: https://www.youtube.com/watch?v=2WGF_ElnkKc

**Reply by nomakewan** [reply]
Don't know what's wrong with your setup, sorry. I personally have Steam, so I can only speak for the experience of users who have RGL in the BTTFV Discord server, not personal experience. All they did was downgrade to 2944, place the contents of the 1.0 version of `GTAV_NoLauncher` into the root directory, and launch the game with the EXE and it worked.

As a note, though, you do not need to ever "downgrade" ScriptHookV. Every version of ScriptHookV works with every version of GTAV (up to whatever the newest SHV version is). Additionally the "patch files" that that person links to are missing the RPFs associated with 3095. Not to mention you only actually need `GTA5.exe` and do not need to downgrade any of the other EXEs.

Also, you _are_ using the ASI for `GTAV_NoLauncher`, right?

**Reply by JokerOmega1** [reply]
Good to know on the ScriptHookV point.

I am indeed using the ASI for GTAV_NoLauncher. As for the RPFs, well I don't have those for 3095, and even though I can't say for sure if having those will help, I doubt it given that this isn't causing a file incompatibility crash or anything like that. I guess I am just screwed, lessoned learned in never purchasing another Rockstar Games product, especially from it's terrible launcher.

Thanks for the help though, it is appreciated, even if it didn't work out.

**Reply by JohnFromGWN** [reply]
> As a note, though, you do not need to ever "downgrade" ScriptHookV. Every version of ScriptHookV works with every version of GTAV (up to whatever the newest SHV version is). Additionally the "patch files" that that person links to are missing the RPFs associated with 3095. Not to mention you only actually need `GTA5.exe` and do not need to downgrade any of the other EXEs.

Not exactly true. While SH5 is designed to be backwards compatible it also updates native functions. The names of these functions change and this can result in code breaking (doesn't affect other mods). The ideal solution is to match SHV5 and GTA5.exe, same versions. I'm not talking hypothetically, this is from experience. Scripts get outdated on a regular basis due to RAGE (the engine) changes.

**Comment by JohnFromGWN** [comment]
There are so many misconceptions about reverting. I've written dozens of tutorials on this subject. You don't need any GTAV_NoLauncher.
1. You don't need to add any mod to revert, all you need is one file. Yes, just one file and that's GTAV.exe.  I know this is contrary to what all the youtube morons advise, but it's just GTAV.exe. Obviously if you're into GTAO, then update update.rpf as well, backing up your customizations first.
2.  If you get the Code 16 error, you will indeed need to kagikn's RSC bypass mod.
3. It is impossible to stop Rockstar from updating. This is another huge misconception. The game will update itself.

Further reading on my blog:

https://gta5modhelp.blogspot.com/2023/05/fix-social-club-code-16-error.html

https://gta5modhelp.blogspot.com/2022/12/avoiding-gta-5-update-headaches_27.html

**Reply by JokerOmega1** [reply]
Unfortunately whenever I copy over an older version of the GTAV.exe, and attempt to run it, I just get the error stating "please run through Rockstar Launcher"...

**Reply by JohnFromGWN** [reply]
Yes, all kinda shit will happen when you take bad advice and/or just fiddle around.
First thing to do is remove any crap like GTAV_NoLauncher mods or even worse cracked .exe files.
I have reverted every single R* update starting with the Contract which broke lots of functionality.
My goto version is 2372 but I also keep every copy of the exe's as well.
Your first step, after removing crap, is to get your game running in vanilla mode. To do this you just rename dinput8.dll to dinput.bak to keep all .asi from running which effectively means no mods and no scripts will work.
If your game doesn't run in vanilla:
1. Do a verify integrity
2. Worst case, and I don't recommend this often, you do a full reinstall
3. If you have trouble with either the RSGL or the RSC, delete them and they will reinstall.
4. Once vanilla is running, rename dinput8.dll. Replace whatever GTA5.exe with the version you want. Ensure your SH5 is the same version or higher. If not, you'll get a reminder message from Mr. Blade

Does this work? I've done it over 20 times for R* updates and also to allow me to run the newest versions without reinstalling or keeping multiple copies of the game. One file, that's all you need.

As for trying to stop GTA V from updating? Only a cracked .exe, possibly infected with malware (detectable or non detectable) is the only way. Also remember if you get a Code 16 error you need the amazing mod from Mr. Kagikn. Having said, I never removed it to see if the new RSC still generates a Code 16 without it.

**Reply by JohnFromGWN** [reply]
P.S. Youtubers, in general, don't know shit about how the game works. Some will even tell you to take dinput8.dll from the System32 folder to fix your game. Not good at all. Stay away from youtube, go back to basics for troubleshooting. I also wrote a guide if you're interested.

https://www.gtainside.com/en/gta5/other/188566-the-gta-5-encyclopedia/

**Reply by nomakewan** [reply]
> There are so many misconceptions about reverting. I've written dozens of tutorials on this subject. You don't need any GTAV_NoLauncher.
> 
>     1. You don't need to add any mod to revert, all you need is one file. Yes, just one file and that's GTAV.exe.  I know this is contrary to what all the youtube morons advise, but it's just GTAV.exe. Obviously if you're into GTAO, then update update.rpf as well, backing up your customizations first.
> 
>     2. If you get the Code 16 error, you will indeed need to kagikn's RSC bypass mod.
> 
>     3. It is impossible to stop Rockstar from updating. This is another huge misconception. The game will update itself.
> 
> 
> Further reading on my blog:
> 
> https://gta5modhelp.blogspot.com/2023/05/fix-social-club-code-16-error.html
> 
> https://gta5modhelp.blogspot.com/2022/12/avoiding-gta-5-update-headaches_27.html

Interesting, but flies in the face of my own personal experience.

1. I just tested your method to downgrade to 1.0.2944.0. I left `update.rpf` and `update2.rpf` as their 1.0.3274.0 versions and just replaced `GTA5.exe`. This resulted in an infinite "Entering Story Mode" screen. Swapping `update.rpf` for the 1.0.2944.0 version but leaving `update2.rpf` alone resulted in a black screen and no load. Swapping `update2.rpf` for the 1.0.2944.0 version finally allowed the game to successfully load into 1.0.2944.0. So I must disagree with you that `GTA5.exe` is the only file necessary to downgrade the game. The only EXE, sure. But not the only file.
2. Again, when our downgrade instructions only included downgrading `GTA5.exe` and the two RPFs, this worked perfectly for our Steam and Epic users. However, all the RGL users complained that whenever they tried to start the game, RGL would automatically update their games back to 1.0.3274.0. This was confirmed experimentally; I had a user downgrade the files alongside me, then they ran RGL, the game loaded, and the resulting `ScriptHookV.log` file had the line `INIT: Success, game version is VER_1_0_3274_0`. Another RGL user then posted their success with using version 1.0 of `GTAV_NoLauncher`, and we added that as an additional step for RGL users. This resolved the issue for the RGL users.

Would like to have your input on the above.

**Reply by JohnFromGWN** [reply]
I don't know what you're doing because you're playing with multiple variables. Go 1 variable at a time. The more changes you make, the greater the chances of bricking your game (pardon the hardware analogy) or even corrupting game files.

This is my process for Steam.
1. My GTA5.exe is 2372 as a default
2. My root folder (ie. everything except mods) is fully updated to 3274
3. My Update.rpf ( dont care about update2.rpf) in the Mods folder is 2372 - you can't mix and match
4. I never touch the RSGL or the RSC - they update themselves

On the now defunct GTA5mods forums I walked a ton of ppl through this process and never heard back any issues. Obviously users who copy over Update.rpf to their Mods folder are destroying all their customized files - the dlclist.xml and most importantly the game will crash because you now have a mismatch between your gameconfig.xml.  If you have 2372, you need its gameconfig, not the one for 3274 and vice versa.

In a properly configured GTA V mod environment, assuming a proper mod load and the proper hardware, any crashes or hanging or other fails are most likely due to improper gameconfig.xml.  And ofc, if you push the game too far, too many  mods, or shitty scripts you can expect fails as well. 

From your description, almost sure its either a mismatch between update.rpf (mods version) and .exe. 
Or it's the wrong gameconfig.xml mod.

One final point. The gameconfig.xml is not a magic bullet. You need to experiment, at times, to ensure you have the proper one not only for your version but for your assets. Modding is part art, and part science, but the process I described, if attention to detail prevails, is foolproof. One of the few elements of modding you can count on to be 100% consistent. Sure, Rockstar can always end the process like when we experienced the code 16 but that wasn't a reverting issue. It was a social club issue and thankfully it was bypassed.

**Reply by nomakewan** [reply]
Okay, so then you do admit that `update.rpf` must match the EXE version, which already flies in the face of your above recommendation suggesting that `GTA5.exe` is the only necessary component for a successful downgrade.

As to the `gameconfig.xml`, the update.rpf from 2944 was using the `gameconfig.xml` for 2944 from pmwparksfan. This resulted in the black screen and no load until the 2944 `update2.rpf` was also included. The only mod installed was v2.3.2 of BTTFV and its dependencies (LemonUI, HeapAdjuster). And, again, requiring a `gameconfig.xml` that matches the EXE version is yet another downgrade requirement that you did not specify (but that, again, BTTFV's downgrade guide also handles).

It just seems like maybe you've been doing this for so long that you're forgetting some of the nitty-gritty details.

**Reply by JohnFromGWN** [reply]
Wow. That's quite a wall of text you threw there.
Let's try to simplify.
I'm on version X of GTA V. Rockstar introduces version Y.
I don't touch my mod version of Update.rpf. This ensures no mismatch and all my customized files including dlclist and gameconfig remain intact.
I get my X version of GTA5.exe and overwrite the new GTA5.exe and that's the only file I will need.

I'm now reverted with one file. I'll repeat one file.

Ok, now I have no idea why you're pushing these arguments. I was trying to help the OP who obviously was having issues.
Your arguments remind me of the countless times i''ve heard this: My game won't run even though I have all the necessary files installed and I followed the instructions, right?
Nope wrong, you screwed up somewhere.

End of discussion on my part.

**Reply by nomakewan** [reply]
John,

Your argument however centers on the user you are addressing already having the older version of the files to choose from. In OP's case, and in the case of many of BTTFV's users we are supporting, that is _not_ the case. Nor should it be _assumed_ that is the case when offering support.

Hopefully that helps clarify things. Have a great one.
